scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/db/TaskDBUtils.java

```diff
@@ -184,10 +189,23 @@ private static StringBuilder getQueryFilteringExpression(boolean hasUser, boolea
 
         result.append("and taskStatus in (:taskStatus) ");
 
+        if (sortParams != null && !sortParams.getSortParameters().isEmpty()) {
+            result.append("order by ");
+            List<SortSpecifierContainer.SortSpecifierItem> items = sortParams.getSortParameters();
+            for (int i = 0; i < items.size(); i++) {
+                SortSpecifierContainer.SortSpecifierItem item = items.get(i);
+                String order = "ascending".compareTo(item.getOrder().toString()) == 0 ? "ASC" : "DESC";
```

Comment:
The default value in `SortSpecifierContainer.SortSpecifierItem` for `order` is *ASCENDING*. Here you check for *ascending* in lower case only. It would be nice to extract allowed values as constants and reuse these constants for performing the check. Optionally, `equalsIgnoreCase` could be used.

Would be even more clear to extract the logic in a dedicated method.